viewing_party/party.py

```diff
@@ -1,23 +1,257 @@
+                        movie_recs.append(movie)
+    
+    return movie_recs
+
 # -----------------------------------------
 # ------------- WAVE 5 --------------------
 # -----------------------------------------
 
+def get_new_rec_by_genre(user_data):
+    # New dict will hold the most watched genre
+    user_watched_list = user_data["watched"]
+    most_watched_genre = {}
+    for movie in user_watched_list:
+        value_genre = movie["genre"]
+        if value_genre in most_watched_genre:
+            most_watched_genre[value_genre] += 1
+        else:
+            most_watched_genre[value_genre] = 1
+
+    # Max_watched_genre hold the value of the most watched genre by user
+    max_watched_genre = ""
+    max_watched = 0
+    for genre, num in most_watched_genre.items():
+        if num > max_watched:
+            max_watched = num
+            max_watched_genre = genre
+    
+    # Create a dict that'll hold the titles of user's watched movies
+    user_watched_movies = {}
+    for movie in user_watched_list:
+        key = movie["title"]
+        user_watched_movies[key] = 1
+
+    # List of recommended movies by most watched genre and user hasn't watched it.
+    rec_movies_by_gender = {}
+    friends_list = user_data["friends"]
+    for dict in friends_list:
+        for value in dict.values():
+            for movie in value:
+                watched_by_friend = movie["title"]
+                genre_movie_friend = movie["genre"]
+                # La agrego si yo no la he visto y si es del genero que mas veo
+                if watched_by_friend in rec_movies_by_gender:
+                    continue
+                if watched_by_friend not in user_watched_movies and genre_movie_friend == max_watched_genre:
+                    rec_movies_by_gender[watched_by_friend] = movie
+
+    return list(rec_movies_by_gender.values())
+
+
+def get_rec_from_favorites(user_data):
+
+    # List that holds the titles of the favorite movies.
+    favorites_list = user_data["favorites"]
+    friends_titles = []
+
+    # Access the list of movies that friends' have watched
+    friends_watched_list = user_data["friends"]
+
+    # Compare every item in the favorites list against the movies in friends_watched_list
+    # Nested loops to access every title in friends_watched_list
+    result_favs = []
+    for dict in friends_watched_list:
+        for value in dict.values():
+            for movie_watched in value:
+                title = movie_watched["title"]
+
+                if title not in friends_titles:
+                    friends_titles.append(title)
+
+    # Loop through Favorites list to compare titles
+    for movie in favorites_list:
+        favorite_title = movie["title"]
+        # Check if the movie is in the user's favorites and None of its friends have watched
+        if favorite_title not in friends_titles and movie not in result_favs:
+            result_favs.append(movie)            
+                
+    return result_favs

```

Comment:
Nice work! How might this function change if we used `get_friends_unique_watched` and `get_most_watched_genre` as helper functions?